app/services/ui_notifications/rss_notifications_checker.rb

```diff
@@ -0,0 +1,50 @@
+require 'rss'
+require 'date'
+
+module UINotifications
+  class RssNotificationsChecker
+    def initialize(options = [])
+      @url = options[:url] || 'https://theforeman.org/feed.xml'
+      @latest_posts = options[:latest_posts] || 3
+      @force_repost = options[:force_repost]
+      @audience = options[:audience] || Notification::AUDIENCE_GLOBAL
+    end
+
+    def deliver!
+      feed = RSS::Parser.parse(@url, false)
+      feed.items[0, @latest_posts].each do |item|
+        blueprint = rss_notification_blueprint
+        if notification_already_exists?(item)
+          next unless @force_repost
+        end
+        initiator = User.anonymous_admin
+        notification = Notification.new(
+          :initiator => initiator,
+          :audience => @audience,
+          :message => item.summary.content,
+          :notification_blueprint => blueprint,
+          :actions => {
+            :links => [
+              {
+                :href => item.link.href,
+                :title => _('Link to blog')
+              }
+            ]
+          }
+        )
+        blueprint.save
```

Comment:
This is a leftover from the changes I made to adapt to UINotifications, the `.save`call won't do anything as the blueprint attributes aren't modified, but it's indeed useless so I'll remove it